nautobot/core/management/commands/generate_performance_test_endpoints.py

```diff
@@ -0,0 +1,143 @@
+                    self.app_name_to_urls["endpoints"][view_name].append(url_pattern)
+                    total_count = len(model_class.objects.all())
+                    page_query_parameter = 5
+                    per_page_query_parameter = total_count // page_query_parameter
+                    # One endpoint with default pagination
+                    self.app_name_to_urls["endpoints"][view_name].append(
+                        url_pattern
+                        + "?per_page="
+                        + str(per_page_query_parameter)
+                        + "&"
+                        + "page="
+                        + str(page_query_parameter)
+                    )
+            else:
+                # TODO handle the case where there is no instances of the model is found
+                self.std.out.write(f"No instances of {model_class} found")
+        else:
+            # A generic endpoint like `core:home`
+            if view_name not in self.app_name_to_urls["endpoints"]:
+                self.app_name_to_urls["endpoints"][view_name] = []
+            self.app_name_to_urls["endpoints"][view_name].append(url_pattern)
+
+    def fetch_urls(self, url_patterns):
+        """
+        Store the URL patterns in the dictionary to output an .YAML file
+        The dictionary will have the following structure:
+        {
+            "endpoints": {
+                <app_name>:<view_name>: [
+                    <url_pattern_1>,
+                    <url_pattern_2>,
+                ],
+                dcim:device: [
+                    "/dcim/devices/cfbd447f-d563-4fac-bb75-bdda70ab4e80/",
+                    "/dcim/devices/38471bfe-0aca-4e09-b545-b0f90280fb66/",
+                ],
+                ...
+            },
+            ...
+        """
+        for pattern in url_patterns:
+            if hasattr(pattern, "url_patterns"):
+                # If it's a nested URL pattern, recursively list its URLs
+                self.fetch_urls(pattern.url_patterns)
+            else:
+                # TODO need to include Nautobot App endpoints as well at some point
+                if pattern.lookup_str.startswith(
+                    (
+                        "nautobot.circuits.views",
+                        "nautobot.cloud.views",
+                        "nautobot.core.views",
+                        "nautobot.dcim.views",
+                        "nautobot.extras.views",
+                        "nautobot.ipam.views",
+                        "nautobot.tenancy.views",
+                        "nautobot.virtualization.views",
+                        "nautobot.wireless.views",
+                    )
+                ):
+                    # Only fetch urls from relevant apps
+                    app_name = pattern.lookup_str.split(".")[1]
+                    model = pattern.default_args.get("model", None)
+                    if model:
+                        app_name = model._meta.app_label
```

Comment:
Some of the view endpoints (changelog, notes) that are still using legacy views (ObjectNotesView and ObjectChangelogView) will yield the app_name `extras` from line 130 even tho the model is from a different app ("ipam:service" for example). So lines 131 to 133 is an attempt to correct that if a model_class is available.